### PR TITLE
Read, write User schoolId

### PR DESCRIPTION
**Pull request by aaronschachter**
### What does this PR do?

This PR starts on reading and writing the current user's school from the `SchoolFinder` component:

* Splits `SchoolFinder` component into separate `UpdateSchool` and `CurrentSchool` components.
* Displays `CurrentSchool` component on page load if user school is already selected
* Displays `UpdateSchool` component if user school is not set, displays `CurrentSchool` upon save

### Any background context you want to provide?

Not displaying school name/info yet, as the `SchoolSelect` component is still querying the lofischools app. Once we get the schools DB + GraphQL integration sorted out (hopefully in https://github.com/DoSomething/graphql/pull/155), I'll make the change to query schools from GraphQL in a future PR.

Won't be adding screenshots, as this feature is still under active development. There aren't any visual changes from #1668.

### What are the relevant tickets/cards?

* Update user school https://www.pivotaltracker.com/story/show/169549547
* Display saved user school https://www.pivotaltracker.com/n/projects/2401401/stories/169549488

### Checklist

* [ ] Documentation added for new features/changed endpoints.
* [ ] Added appropriate feature/unit tests.

### CODE REVIEW

File: resources/assets/components/utilities/SchoolFinder/UpdateSchool.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+          />
+        </div>
+      ) : null}
+      <Mutation
+        mutation={USER_SCHOOL_MUTATION}
+        variables={{ schoolId, userId }}
+        update={() => onSubmit(schoolId)}
+      >
+        {userMutation => (
+          <Button onClick={userMutation} disabled={schoolId === null} attached>
```
Comment (mendelB) [inline resources/assets/components/utilities/SchoolFinder/UpdateSchool.js]: Could this just be `disabled={!schoolId}` to account for an `undefined`/omitted prop?

File: resources/assets/components/utilities/SchoolFinder/SchoolFinder.js [hunk 2]
```diff
@@ -1,85 +1,57 @@
...
-            <h3>{selectedSchool.name}</h3>
-            <small className="uppercase">
-              {selectedSchool.city}, {selectedSchool.state}
-            </small>
-          </div>
-        </Card>
-      </div>
-    );
-  }
+  const [currentSchool, setCurrentSchool] = useState(null);
```
Comment (lindsayrmaher) [inline resources/assets/components/utilities/SchoolFinder/SchoolFinder.js]: I noticed this syntax in a few places on the PR, just wanted to get clarity on what exactly it's doing!

Comment (mendelB) [reply resources/assets/components/utilities/SchoolFinder/SchoolFinder.js]: It's React Hooks!! ⚓️ 

Their [Hooks at a Glance](https://reactjs.org/docs/hooks-overview.html) doc is a great place to get a quick primer on the basic syntax, but the overall documentation is excellent

File: resources/assets/components/utilities/SchoolFinder/UpdateSchool.js (hunk 1 above)
Comment (aaronschachter) [reply resources/assets/components/utilities/SchoolFinder/UpdateSchool.js]: Good call. This is about to change up now that we have Schools in GraphQL -- so I'll address in a future PR.

File: resources/assets/components/utilities/SchoolFinder/SchoolFinder.js (hunk 2 above)
Comment (aaronschachter) [reply resources/assets/components/utilities/SchoolFinder/SchoolFinder.js]: These are [State Hooks](https://reactjs.org/docs/hooks-state.html), which are a fun very-new way to get/set a component's state.

Comment (mendelB) [reply resources/assets/components/utilities/SchoolFinder/SchoolFinder.js]: But tl;dr the `useState(null)` is the same as `this.state = {currentSchool: null}`
it returns as array with the first value being the state reference `this.state.currentSchool === currentSchool` and the second being the update method `this.setState({currentSchool: "123"} === `setCurrentSchool("123")`)

**Pull request merged**